local.tf

```diff
@@ -0,0 +1,19 @@
+locals {
+  host_pool_name                 = var.host_pool_name_override == "" ? "${var.system_name}-hostpool" : var.host_pool_name_override
+  workspace_name                 = var.workspace_name_override == "" ? "${var.system_name}-workspace" : var.workspace_name_override
+  desktop_application_group_name = var.desktop_application_group_name_override == "" ? "${var.system_name}-appgroup" : var.desktop_application_group_name_override
+
+  host_pool_load_balancer_type = var.host_pool_type == "Personal" ? "Persistent" : var.host_pool_load_balancer_type
+
+  custom_rdp_properties = join(";",
+    flatten([
+      var.custom_rdp_properties,
+      var.aad_joined_allow_access_from_nonjoined ? ["targetisaadjoined:i:1"] : [""]
```

Comment:
I suggest adding booleans for managing other default RDP properties, e.g. disk or audio redirects.
https://docs.microsoft.com/en-us/azure/virtual-desktop/customize-rdp-properties#default-rdp-file-properties

This is probably out of scope for this PR.

Reply:
Good idea. I'm not too happy about the readability of the code around how the custom RDP properties turned out. In my opinion there is too much going on in the code you referenced. In addition there is some logic in the host-pool resource itself.
I raised as suggested #6.